### PR TITLE
Adding demo scripts as basic example to follow of the structure when developing models.

### LINKED ISSUE #33: Creating Autoencoder Example
The goal is to adapt the following script created by @thogar-computer [train_eurosat.py
](https://github.com/NERC-EDS/FRAME-FM/blob/757398a3b39508ed536897110e34530d6b2c5f75/train_eurosat.py) to follow an CAE (convolutional autoencoder) framework rather than just a CNN framework. The inspiration for the CAE framework is being taken from this paper: [An autoencoder driven deep learning geospatial approach to flood vulnerability analysis in the upper and middle basin of river Damodar](https://www.nature.com/articles/s41598-025-96781-2#Sec21). The steps involved to resolve this are as follows:

- [ ] Change resnet model for custom encoder script.
- [ ] Include mirror of this as the decoder.
- [ ] Write a function for evaluating a cost function with respect to the difference to the input data.
- [ ] Wrap the whole script in the relevant wrappers being created by @asw-v4 [Shared utilities used across the project, such as I/O helpers, metrics, and transforms.](https://github.com/NERC-EDS/FRAME-FM/blob/623ebb2ae30d7a83075a3283a0b0d7a5fc486a89/src/FRAME_FM/utils/utils_README.md)
- [ ] Evaluate the model performance and examine the latent space generated (visualising clusters etc)

This work will be used to help provide a basis for the exemplars for WP4. It will also help inform decisions around data loaders and computational infrastructures.

### PR DESCRIPTION
**Pull request by Jez-Carter**
Resolves #33. Demo scripts are added that use [EuroSAT data from Torchvision](https://docs.pytorch.org/vision/main/generated/torchvision.datasets.EuroSAT.html) and fit a basic autoencoder using the MSE loss of the reconstructed input to the original input. The aim is to simply demonstrate the approximate structure for future scripts to follow. These demo scripts will be replaced with ones that use data specific to the project partners. 

Additional work is needed to:
- Add some documentation around how to run the demo (go to root of folder and run something like `python src/FRAME_FM/training/train.py   data.train_split=0.05 data.val_split=0.01 data.test_split=0.0` - where the kwargs are for speedup).
- To demonstrate how to examine the output and performance metrics etc.

### CODE REVIEW

**Comment by Jez-Carter** [comment]
> @Jez-Carter, I can see you have renamed the config files. The idea with Hydra is that we can have multiple config files in the directory and update either the main config defaults or override them on the command line.
> 
> As this is the start, and we are getting a working demo up and running, replacing the current one isn't a massive issue

I thought it would be good just to keep it clean, I know hydra allows multiple config files but I didn't think it made sense to leave ones that have been superseded. As in, at least in the repo main, all configs should be with respect to showing something specific and not different versions of the same task.

**Comment by Jez-Carter** [comment]
I've made a couple extra commits that make the following changes:

- In demo_eurosat.yaml I've changed made the following adjustment `data_root: ${oc.env:DATA_ROOT,${hydra:runtime.cwd}/data}`. This means data downloaded for the run is kept in the /data folder of the root of the directory, unless the DATA_ROOT env variable is set, which can be useful if you want it to point at some shared folder location on JASMIN for example. I've done something similar for the demo_mlflow.yaml file, setting `tracking_uri: ${oc.env:MLFLOW_TRACKING_URI,${hydra:runtime.cwd}/mlruns}`.
- I've renamed the files to more closely reflect the normal pattern, so for example: instead of demo_data.yaml it's now demo_eurosat.yaml as in the future we expect people to add config files named by the dataset they address, e.g. era5.yaml 

I think it's ready for merging, @thogar-computer are you happy?
